File: BOJ/CPP/단계별로_풀어보기/04_1차원_배열/BOJ_8958.cpp
/*
	BOJ 8958 : OX퀴즈

	author : kdgyun
	link https://st-lab.tistory.com
	link https://github.com/kdgyun
	Created by kdgyun
	Copyright © 2021 kdgyun. All rights reserved.
*/

#include <iostream>
#include <string>

using namespace std;

int OX(string& s) {
	
	int cumulative = 0;
	int sum = 0;

	for(char &v : s) {
		/*
		 * O 문자일 경우 누적합을 1 증가시킨 뒤 
		 * 해당 값에 대해 누적합
		 */
		if(v == 'O') {
			cumulative++;
			sum += cumulative;
		}
		else {
			cumulative = 0;
		}
	}
	return sum;
}

int main(int argc, const char * argv[]) {

	ios_base::sync_with_stdio(0);
	cin.tie(nullptr);

	int T;
	cin >> T;

	for(int i = 0; i < T; i++) {
		string s;
		cin >> s;

		cout << OX(s) << "\n";
	}

	return 0;
}

File: BOJ/Java/단계별로_풀어보기/B_04_1차원_배열/BOJ_8958.java
package 단계별로_풀어보기.B_04_1차원_배열;

/**
 * BOJ 8958 : OX퀴즈
 * 
 * @author kdgyun 
 * {@link https://st-lab.tistory.com}
 * {@link https://github.com/kdgyun}
 *
 */

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class BOJ_8958 {
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		StringBuilder sb = new StringBuilder();

		int T = Integer.parseInt(br.readLine()); // 테스트 케이스

		for (int i = 0; i < T; i++) {

			int cnt = 0; // 연속횟수
			int sum = 0; // 누적 합산

			for (byte value : br.readLine().getBytes()) {
				
				if (value == 'O') {
					cnt++;
					sum += cnt;
				} 
				else {
					cnt = 0;
				}
			}
			
			sb.append(sum).append('\n');

		}
		System.out.println(sb);

	}
}
